config/initializers/stripe.rb

```diff
@@ -1 +1,14 @@
-Stripe.api_key = ENV['stripe_secret_key']
\ No newline at end of file
+Stripe.api_key = ENV['stripe_secret_key']
+
+StripeEvent.configure do |events|
+  events.subscribe 'charge.succeeded' do |event|
+    customer = User.where(customer_token: event.data.object.customer).first
+    Payment.create(user: customer, amount: event.data.object.amount, reference_id: event.data.object.id)
+  end
+
+  events.subscribe 'charge.failed' do |event|
+    customer = User.where(customer_token: event.data.object.customer).first
```

Comment:
Homework for you: `find_by` is better than `where().first`, find out why. You may check on lesson 1 too and its assignments, there is a note that may hint it out.